DemoApp/Sources/Views/Controls/DemoControls.swift

```diff
@@ -164,3 +165,31 @@ extension View {
         }
     }
 }
+
+struct SnapshotIconView: View {
+
+    @Injected(\.images) var images
+    @Injected(\.snapshotTrigger) var snapshotTrigger
+
+    let size: CGFloat
+
+    init(size: CGFloat = 44) {
+        self.size = size
+    }
+
+    var body: some View {
+        Button(
+            action: {
+                snapshotTrigger.capture = true
```

Comment:
I think this is a bit error prone atm - you can forget to set it to `false` for example. Ideally, this would be a method `capture()`, which will handle all the manipulation of the state. Although it's a demo app implementation, I think it's good if we improve this, wdyt?

Reply:
The reason this is like that is because the control and actually viewHierarchy we need to snapshot are in different places. Without making changes in our UI infra i couldn't figure out a way to do it. Do you have any suggestions?

Reply:
Yes, it's a bit tricky with the current setup. One idea could be: maybe we change `captureTrigger: AnyPublisher<Bool, Never>` to be a `@Binding`, since it's used in a SwiftUI view? That way when the snapshot is generated, we just reset it to `false` - similar to how `NavigationLink` works.

Reply:
But then someone can forget to set it to false and that would bring us to the same situation, right? The difference with the publisher is that it's a stream and doesn't affect rendering. While changing the value of a binding will cause re-renders.

Reply:
If it's binding, we can change it directly in `SnapshotViewContainer`, when the snapshot is generated, right? Re-renders are not a problem here, this value will change only on user action.

Reply:
The problem with the re-renders is that we end up capturing an empty view (even with a small delay).